### ISSUE TITLE
6581 Filter Bug

### ISSUE THREAD

**Issue by acrouzet** [libsidplayfp/libsidplayfp#80]
The 6581 filter emulation can produce undertones that I don't think occur on real 6581s. This is especially noticeable when Mitch & Dane's tune "Listen and Learn" is being played back with a filter curve setting of 0.9. The undertones in question start on channel 1 at around 1:44, and are an octave below the frequency that's supposed to be output.

Here's a comparison. First is a real 6581, second is sidplayfp:
https://cdn.discordapp.com/attachments/535253084104949760/1113660662812508212/Listen_and_Learn_comparison.wav

**Comment by drfiemost** [comment]
Guess it's something similar to what happens with Wizball (#2)

**Comment by reFX-Mike** [comment]
And in my opinion, it's impossible to fix, since your 6581 might be completely different than whatever "Mitch & Dane's" tune used. It was probably also a 6581, but who knows how their analog filter behaves compared to yours?

I truly think we have to live with the fact that there is no perfect solution for this.

You can't expect this project to tweak its filter settings until it sounds perfect compared to your specific 6581, because then it will not be perfect for the other 99.9% of 6581 out there.

The reality is that there are no perfect settings.

That's why I suggested the building of a database with settings for the most popular composers, so that we get at least "extremely good sounding".

**Comment by acrouzet** [comment]
I'm aware that the filters can sound wildly different between chips, but I don't believe that this "octave drop" effect occurs in this case on any 6581 chips, which is the issue I'm referring to here.

**Comment by reFX-Mike** [comment]
We recently discussed another filter adjustment, different than the filter curve setting you’ve used. Do you hear the same octave drop with the filter curve at its default? If not, then the new setting might bring filter sound in line with what you would expect. I can render an example after Christmas, as I’m currently rather busy.

**Comment by acrouzet** [comment]
It didn't occur. However, what I'm saying is that this issue shouldn't occur on any filter curve setting.

**Comment by reFX-Mike** [comment]
Well, the filter curve setting should probably be restricted to a narrower ranger and the new adjustment, you don’t have access to at the moment, will shift the cutoff without causing the issue you’re experiencing.

In my experiments with the new parameter, I leave the existing filter curve parameter at its default and still get extreme filter curve differences without introducing these kind of artifacts. It’s amazing. Just give me a few days, and after Christmas I can send you render of that tune with adjusted settings.

**Comment by reFX-Mike** [comment]
Here is a quick demo.

https://youtu.be/GPq1geipu5c

You can also see the activity of each SID voice, the filter-state, etc., to get a better understanding of what is going on.

**Comment by acrouzet** [comment]
Nice interface, is this your program?

**Comment by reFX-Mike** [comment]
Yep, it also has CRT emulation for loading/title screens, playlists, built in filter tables for most popular composers, optional audio FX (turned off for this video), a volume database so all tunes play at the same perceived volume, and a few more things I forgot.

Should be finished soonish. It’s using a modified libsidplay and will be open source.

**Comment by acrouzet** [comment]
Some news on this matter: I've been talking to the lead developer of the new Plogue Chipsynth SID emulation, and he confirms that this octave drop effect can occur on real 6581s.

He explains: "That pitch-related phenomenon is something analog filters sometimes do; Every time the input wave cycles, the filter can take a different path, and if it alternates between 2 different paths, you get an additional pitch one octave down. (in layman's terms)"

This might be something that only occurs on certain 6581s, since I've never heard it happen on mine or the Stone Oakvalley chips. 

Since residfp seems to have consolidated all of the filter curve parameters into one, I assume that the goal for this emulation engine is to have the least amount of adjustment needed to get an adequate emulation of most 6581s out there. I think testing many 6581 chips is the only real solution to determining when, or if, the octave drop effect should occur, which of course is a difficult task. However, in the vast majority of the chips emulated in Plogue's engine, this octave drop effect doesn't seem to occur, although this is still an emulation.

**Comment by drfiemost** [comment]
Would be interesting to find one chip affected with the octave drop. I've tried one 6581 marked 3884 and Wizball sounds good, while the Arkanoid glitch is present (#66).

As for the parameters, in 2.7 there will be another one added (see #101) which, according to [reFX-Mike](https://github.com/reFX-Mike), provides even better configurability.
